application/frontend/controllers/AuthController.php

```diff
@@ -128,6 +164,65 @@ public function getAfterLoginUrl($returnTo)
         } else {
             $path = '';
         }
-        return \Yii::$app->params['ui_url'] . $path;
+        return \Yii::$app->params['uiUrl'] . $path;
+    }
+
+    /**
+     * Get client_id from request or session and then store in session
+     * @return string
+     * @throws BadRequestHttpException
+     */
+    public function getClientIdOrFail()
+    {
+        $clientId = \Yii::$app->request->get('client_id');
+        if ($clientId === null) {
+            $clientId = \Yii::$app->session->get('clientId');
+            if ($clientId === null) {
+                throw new BadRequestHttpException('Missing client_id');
+            }
+        }
+        \Yii::$app->session->set('clientId', $clientId);
```

Comment:
Something about the order in which we handle the client id here (check the request, then check the session, then if we found it in either place we store it back into the session) seems off to me, but I can't quite put my finger on why. Anyone else have any misgivings about this part of the code?